client/ayon_core/plugins/publish/collect_otio_frame_ranges.py

```diff
@@ -1,93 +1,204 @@
         otio_tl_range = otio_clip.range_in_parent()
-        otio_src_range = otio_clip.source_range
-        otio_avalable_range = otio_clip.available_range()
-        otio_tl_range_handles = otio_range_with_handles(
-            otio_tl_range, instance)
-        otio_src_range_handles = otio_range_with_handles(
-            otio_src_range, instance)
+        otio_tl_range_handles = otio_range_with_handles(otio_tl_range, instance)
+
+        # Convert to frames
+        tl_start, tl_end = otio_range_to_frame_range(otio_tl_range)
+        tl_start_h, tl_end_h = otio_range_to_frame_range(otio_tl_range_handles)
 
-        # get source avalable start frame
-        src_starting_from = otio.opentime.to_frames(
-            otio_avalable_range.start_time,
-            otio_avalable_range.start_time.rate)
-
-        # convert to frames
-        range_convert = otio_range_to_frame_range
-        tl_start, tl_end = range_convert(otio_tl_range)
-        tl_start_h, tl_end_h = range_convert(otio_tl_range_handles)
-        src_start, src_end = range_convert(otio_src_range)
-        src_start_h, src_end_h = range_convert(otio_src_range_handles)
         frame_start = workfile_start
         frame_end = frame_start + otio.opentime.to_frames(
             otio_tl_range.duration, otio_tl_range.duration.rate) - 1
 
-        # in case of retimed clip and frame range should not be retimed
-        if workfile_source_duration:
-            # get available range trimmed with processed retimes
-            retimed_attributes = get_media_range_with_retimes(
-                otio_clip, 0, 0)
-            self.log.debug(
-                ">> retimed_attributes: {}".format(retimed_attributes))
-            media_in = int(retimed_attributes["mediaIn"])
-            media_out = int(retimed_attributes["mediaOut"])
-            frame_end = frame_start + (media_out - media_in) + 1
-            self.log.debug(frame_end)
-
         data = {
             "frameStart": frame_start,
             "frameEnd": frame_end,
             "clipIn": tl_start,
             "clipOut": tl_end - 1,
             "clipInH": tl_start_h,
             "clipOutH": tl_end_h - 1,
+        }
+        instance.data.update(data)
+        self.log.debug(f"Added frame ranges: {pformat(data)}")
+
+
+class CollectOtioSourceRanges(pyblish.api.InstancePlugin):
+    """Collect source media frame ranges from OTIO clip.
+
+    This plugin extracts and stores source media frame ranges including
+    handles from the OTIO clip.
+
+    Requires:
+        otioClip (otio.schema.Clip): OTIO clip object
+
+    Provides:
+        sourceStart (int): Source media start frame
+        sourceEnd (int): Source media end frame
+        sourceStartH (int): Source media start frame with handles
+        sourceEndH (int): Source media end frame with handles
+    """
+
+    label = "Collect Source OTIO Frame Ranges"
+    order = pyblish.api.CollectorOrder - 0.07
+    families = ["shot", "clip"]
+    hosts = ["hiero", "flame"]
```

Comment:
Shouldn't we instead use different family from product types to filter instances to avoid `hosts` filtering?

Reply:
Not sure this is part of the scope of this PR. Also not up to me to decide how the workflow is designed.

Reply:
I am assuming it would be better to create Issue with redesign suggestion.

Reply:
We're adding more and more plugins making it more and more hard to achieve that. Plugins in ayon-core should not have any filters in `hosts`.